src/components/Popup/Popup.tsx

```diff
@@ -96,56 +109,97 @@ export default class Popup extends UIComponent<Extendable<IPopupProps>, IPopupSt
     as: Portal,
     align: 'start',
     position: 'above',
+    accessibility: PopupBehavior as Accessibility,
   }
 
-  public state = { triggerRef: undefined }
+  public state = { triggerRef: undefined, popupOpened: false }
+
+  actionHandlers: AccessibilityActionHandlers = {
+    performClick: e => this.onTriggerClick(e),
+    closeAndFocusTrigger: e => this.handlePopupState(e, false, this.focusTrigger),
+  }
 
   public renderComponent({
     ElementType,
     classes,
+    accessibility,
     rest,
     rtl,
   }: IRenderResultConfig<IPopupProps>): React.ReactNode {
-    const { children, trigger } = this.props
+    const { children, trigger, position, align } = this.props
 
     return (
       <ElementType
         className={classes.root}
         {...rest}
+        open={this.state.popupOpened}
         trigger={trigger}
         triggerRef={this.handleTriggerRef}
+        triggerAccessibility={{
+          ...accessibility.attributes.trigger,
+          ...accessibility.keyHandlers.trigger,
+        }}
+        onOutsideClick={e => this.handlePopupState(e, false)}
+        onTriggerClick={this.onTriggerClick}
       >
-        {childrenExist(children) ? children : this.renderContent(rtl)}
+        {childrenExist(children)
+          ? children
+          : this.renderContent(
+              computePopupPlacement({ align, position, rtl }),
+              this.renderPopperChildren.bind(this, rtl, accessibility),
+            )}
       </ElementType>
     )
   }
 
-  private renderContent(rtl: boolean): JSX.Element {
-    const { align, position } = this.props
+  private renderContent(
+    popupPlacement: Placement,
+    renderChildrenCb: (props: PopperChildrenProps) => React.ReactNode,
+  ): JSX.Element {
     const triggerRef = this.state.triggerRef
-    const placement = computePopupPlacement({ align, position, rtl })
 
     return (
       triggerRef && (
         <Popper
-          placement={placement}
+          placement={popupPlacement}
           referenceElement={triggerRef}
-          children={this.renderPopperChildren.bind(this, rtl)}
+          children={renderChildrenCb}
```

Comment:
would suggest to rename it to something like `renderPopperChildrenFromProps`, this should prevent some sources of confusion (like the ones that @mnajdova has experienced while writing here comment, I felt the same ones)

Reply:
agree, will rename it